src/data/abilities/ab-attrs/post-summon-copy-ability-ab-attr.ts

```diff
@@ -14,37 +13,24 @@ export class PostSummonCopyAbilityAbAttr extends PostSummonAbAttr {
   private target: Pokemon;
   private targetAbilityName: string;
 
-  public override apply(pokemon: Pokemon, simulated: boolean): boolean {
-    const targets = pokemon.getOpponents();
-    if (!targets.length) {
-      return false;
-    }
-
-    let target: Pokemon;
-    if (targets.length > 1) {
-      globalScene.executeWithSeedOffset(() => (target = randSeedItem(targets)), globalScene.currentBattle.waveIndex);
-      target = target!;
-    } else {
-      target = targets[0];
+  public override apply(pokemon: Pokemon, simulated: boolean): void {
+    if (!simulated) {
+      this.targetAbilityName = allAbilities[this.target.getAbility().id].name;
+      pokemon.summonData.ability = this.target.getAbility().id;
+      this.target.waveData.abilitiesRevealed.push(this.target.getAbility().id);
+      pokemon.updateInfo();
     }
```

Comment:
Reduces `.getAbility()` calls from 3 to 1, and removes the unnecessary `allAbilities[abilityId]`.
```suggestion
    if (simulated) {
      return;
    }

    const targetAbility = this.target.getAbility();
    this.targetAbilityName = targetAbility.name;
    pokemon.summonData.ability = targetAbility.id;
    this.target.waveData.abilitiesRevealed.push(targetAbility.id);
    pokemon.updateInfo();
```